### PR TITLE
tests: Additional tests

### PR DESCRIPTION
**Pull request by adthrasher**
Adding more in-depth checks of file contents, particularly for BAM and FASTQ format output.

### CODE REVIEW

File: tests/tools/test_samtools.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    assert len(read_groups) == 2
+    assert read_groups[0] == "test2"
+    assert read_groups[1] == "test.bwa_aln_pe"
+
+@pytest.mark.workflow('samtools_collate', 'samtools_collate_to_fastq')
+def test_samtools_collate(workflow_dir):
+    bam = pathlib.Path(workflow_dir, 'test-output/out/collated_bam/test.bwa_aln_pe.collated.bam')
+    samfile = pysam.AlignmentFile(bam, "rb")
+
+    reads = list(samfile.head(n=100))
```
Comment (adthrasher) [inline tests/tools/test_samtools.py:40]: I think this BAM is small enough that we could read the entire thing. Is that worth a little runtime penalty? If some rare case caused it to be not correctly `collated`, we probably wouldn't catch it anyway with a BAM this size.

File: tools/samtools.wdl [hunk 2]
```diff
@@ -102,12 +107,34 @@ task split {
...
+                    -h 0 \
+                    -n 1 \
+                    "$out_bam" \
+                    > first_read.sam
+
+                if [ ! -s first_read.sam ]; then
+                    >&2 echo "No reads are in output BAM $out_bam!"
+                    >&2 echo "This is likely caused by malformed RG records."
+                    rm first_read.sam
+                    exit 42
```
Comment (adthrasher) [inline tools/samtools.wdl:133]: Can you modify this to wait and exit after checking all BAMs? I doubt there would be multiple empty read groups, but we should probably catch that case in a single check.

Comment (a-frantz) [reply tools/samtools.wdl:133]: Good idea

File: tests/tools/test_samtools.py (hunk 1 above)
Comment (a-frantz) [reply tests/tools/test_samtools.py:40]: We might as well read the whole thing. I doubt it will be noticeable given the ~15min this already takes.

**Comment by a-frantz** [comment]
@adthrasher the rollback did not fix the CI unfortunately

**Review by a-frantz — APPROVED** [review]
This looks great @adthrasher ! I'll let you merge. We've still got more tests to add, but I think iteratively merging bits when we have bandwidth is going to be better than waiting for all our desired testing to happen at once.

**Pull request merged**